### ISSUE TITLE
Notifications not appearing

### ISSUE THREAD

**Issue by davidemf** [solus-project/budgie-desktop#1472]
I've installed budgie-desktop on an existing ubuntu 18.04 installation. Everything works wonderfully well, but I cannot see notifications. If I do `notify-send "foo" "bar"` it hangs for a while and then returns with no message and success exit code, but again no notification. In the budgie panel the notification tray is also empty.

Any suggestion for what could be wrong and how to test it/fix it?

**Operating System**: Ubuntu 18.04
**Budgie version**: Budgie 10.4

**Comment by JoshStrobl** [comment]
Mind taking a look @fossfreedom ? Not having these issues under Solus.

**Comment by davidemf** [comment]
Many thanks. 
Yes it's Ubuntu with Gnome Shell and notifications do work there. 
I installed it with a simple `apt install ubuntu-budgie-desktop`

**Comment by alunux** [comment]
Not having these issues under Fedora.

**Comment by fossfreedom** [comment]
@alunux similarly - I cannot reproduce here with the OP's setup (Ubuntu 18.04 + budgie separately installed) nor on Ubuntu Budgie.  So just trying a process of elimination.

**Comment by davidemf** [comment]
@fossfreedom, thanks. I've reset the panel as instructed and I /can /see 
the bell icon, but still no notification appearing.

**Comment by fossfreedom** [comment]
@davidemf is the package libnotify-bin installed?

**Comment by davidemf** [comment]
Yes it is.
I've just confirmed that when switching back to gnome desktop (ubuntu 
default) I can see all notifications, and `notify-send foo bar` returns 
immediately and shows a pop-up. Switching back to budgie, `notify-send` 
hangs for 25 seconds and then returns without showing a notification:

```
time notify-send hi ho

real    0m25.026s
user    0m0.004s
sys    0m0.000s
```

I'm also experiencing a temporarily frozen cursor which prevents me from 
typing for a few seconds (it happens often in applications like Slack). 
I suspect that it might be related to the notification problem as it 
only happens occasionally and disappears after a while, but I have no 
hard evidence for the connection between the two.

**Comment by fossfreedom** [comment]
Please can you create a new user account and login with that - does notifications work correctly?  Just trying to see if this is a system wide issue or localised to a specific account.

**Comment by davidemf** [comment]
Fresh account created. Same problem occurs. I've also logged into gnome shell and notifications work correctly there under the new user.

**Comment by davidemf** [comment]
Anything else I could try and report on? At the moment it's the only problem preventing me from using budgie. Many thanks.

**Comment by fossfreedom** [comment]
Can you describe in more detail what is in your installation.  i.e.  have you self compiled any software?  Have you got any PPA's installed such as the GNOME3 Staging PPA?

As mentioned previously - I have the same GNOME-Shell / Budgie combo and no issues - so there must be a conflicting library from software you have installed.  

You similarly can double check that notifications work through the Ubuntu Budgie 18.04 ISO live-session.

**Comment by lvanderwiele** [comment]
I don't know if that helps but I'm currently on Ubuntu 18.04 with Budgie 10.4 (originally an Ubuntu 17.04 with later Shell installed by me and Budgie installed less than a month ago) and I do not have a problem with non-appearing notifications.

That being said, I didn't install using `sudo apt install ubuntu-budgie-desktop` bur rather using `sudo apt install budgie-desktop`. I don't if that will change anything but it might help.

**Comment by chrBrd** [comment]
I'm having similar issues on a clean install of Ubuntu Budgie 18.10. Notification popus appear on-screen as expected, the bell icon lights up to indicate there are unread notifications, but the actual 'Notifications' panel is empty aside from the usual 'No unread notifications' and 'Nothing to see here' messages.

**Comment by eddie-dunn** [comment]
For what it's worth, I have this problem with `dunst` on Solus Linux. I think it occurred after I installed Gnome shell (and GDM), KDE Plasma, and MATE, so I am assuming it's because of some notification service conflict.

Notifications work when log into Budgie desktop.

**Comment by eddie-dunn** [comment]
OK, I've found what causes `notify-send` to hang in my case -- it's the KDE notification service.

`/usr/share/dbus-1/services/org.kde.plasma.Notifications.service` contains the following line:

```
Exec=/usr/bin/plasma_waitforname org.freedesktop.Notifications
```

`plasma_waitforname` won't work outside of KDE.

A quick and dirty workaround is to disable all unwanted notification DBUS services containing `org.freedesktop.Notifications`

They can be found by like so

```
$ grep -r org.freedesktop.Notifications /usr/share/dbus-1/services
/usr/share/dbus-1/services/org.freedesktop.mate.Notifications.service.disabled:Name=org.freedesktop.Notifications
/usr/share/dbus-1/services/org.knopwob.dunst.service:Name=org.freedesktop.Notifications
/usr/share/dbus-1/services/org.kde.plasma.Notifications.service.disabled:Name=org.freedesktop.Notifications
```

Note how I have appended `.disabled` to the notification services I don't want. More info can be found [here](https://askubuntu.com/a/1092324/18380). Unfortunately, as that answer details, it doesn't seem like it's possible to have multiple `org.freedesktop.Notifications` services enabled simultaneously without [cumbersome workarounds](https://askubuntu.com/a/957624/18380).
